src/blockdigest.h

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2017 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef _BITCOIN_SRC_BLOCKDIGEST_H_
+#define _BITCOIN_SRC_BLOCKDIGEST_H_
+
+#include <vector>
+#include <boost/filesystem.hpp>
+
+#include "bloom.h"
+#include "serialize.h"
+#include "uint256.h"
+
+class COutPoint;
+class CBlock;
+class CTransaction;
+
+namespace DigestFilter {
+
+enum format {
+    vectorized      = 1 << 0,   // argument is a uint8_t indicating size
+    mapped          = 1 << 1,   // argument comprises two uint8_t's indicating offset and count
+    inverted        = 1 << 2,   // listed elements are ABSENT; unlisted elements are PRESENT
+};
+
+typedef std::vector<uint8_t> data_t;
```

Comment:
Also prevector is still resizable, you just want to target the right percentile such that it's not a crazy amount of memory, but you capture the size of most of the elements. This can speed up your code pretty largely. Furthermore, you aren't *actually* saving any memory with a 20 byte vector, because a 20 byte vector will malloc 32 bytes under the hood (I'm almost certain).

If you really want to use the minimum storage overhead/runtime overhead, you can make a data structure that is 2 vectors, one raw memory and the other just the pointers into it.